File: app/src/main/java/com/sjn/stamp/db/CategoryStamp.java
package com.sjn.stamp.db;

import io.realm.RealmObject;
import io.realm.annotations.Index;
import io.realm.annotations.PrimaryKey;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

@Accessors(prefix = "m")
@AllArgsConstructor(suppressConstructorProperties = true)
@NoArgsConstructor
@Getter
@Setter
public class CategoryStamp extends RealmObject {

    @PrimaryKey
    public long mId;
    @Index
    public String mName;
    public String mValue;
    public String mType;
}

File: app/src/main/java/com/sjn/stamp/db/CategoryStamp.kt
package com.sjn.stamp.db

import android.support.v4.media.MediaMetadataCompat
import com.sjn.stamp.constant.CategoryType
import com.sjn.stamp.utils.MediaItemHelper
import io.realm.RealmObject
import io.realm.annotations.Index
import io.realm.annotations.PrimaryKey

open class CategoryStamp(
        @PrimaryKey var id: Long = 0,
        @Index var name: String = "",
        var isSystem: Boolean = false,
        var value: String = "",
        var type: String = ""
) : RealmObject() {
    fun contain(mediaMetadata: MediaMetadataCompat): Boolean =
            MediaItemHelper.fetch(mediaMetadata, CategoryType.of(type)) == value
}